### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Rungetf**
* [ ] Flake8 passes (`flake8 . --exclude=.venv,.build,planemo_test_env,build --ignore=E501,F403,E402,F999,F405,E712`)
* [ ] Local tests pass (`py.test hicexplorer --doctest-modules`)

### CODE REVIEW

**Comment by joachimwolff** [comment]
Thanks for the PR @Rungetf! I changed the to be merged branch to develop, and I will do a review today.

@all This is a PR from Frederic a new HiWi in our lab. He will work on test cases to increase the test coverage.

File: .pytest_cache/v/cache/nodeids [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  "hicexplorer/test/test_hicmatrix.py::test_maskBins",
+  "hicexplorer/test/test_hicmatrix.py::test_update_matrix",
+  "hicexplorer/test/test_hicmatrix.py::test_restoreMaskedBins",
+  "hicexplorer/test/test_hicmatrix.py::test_reorderMatrix",
+  "hicexplorer/test/test_hicmatrix.py::test_truncTrans",
+  "hicexplorer/test/test_hicmatrix.py::test_truncTrans_bk",
+  "hicexplorer/test/test_hicmatrix.py::test_removePoorRegions",
+  "hicexplorer/test/test_hicmatrix.py::test_printchrtoremove",
+  "hicexplorer/test/test_hicmatrix.py::test_removeBySequencedCount"
+]
```
Comment (joachimwolff) [inline .pytest_cache/v/cache/nodeids]: Should this file be included or isn't it a local file from you?

File: .travis.yml [hunk 2]
```diff
@@ -55,4 +55,4 @@ install:
 
 # command to run tests
 script:
-  - py.test hicexplorer --doctest-modules
+  - py.test hicexplorer --doctest-modules --capture=sys
```
Comment (joachimwolff) [inline .travis.yml]: Out of interest: What is this option doing?

File: hicexplorer/test/test_hicmatrix.py [hunk 3]
```diff
@@ -77,12 +80,12 @@ def test_convert_to_zscore_matrix():
     # make Hi-C matrix based on test matrix
     hic = hm.hiCMatrix()
     hic.matrix = csr_matrix(mat)
-    cut_intervals = [('chr', idx, idx + 10, 0) for idx in range(0, mat.shape[0] * 10, 10)]
+    cut_intervals = [('chr', idx, idx + 10, 0)
+                     for idx in range(0, mat.shape[0] * 10, 10)]
```
Comment (joachimwolff) [inline hicexplorer/test/test_hicmatrix.py]: Can you write it one line? It should be possible to deactivate the linting rule here.

File: hicexplorer/test/test_hicmatrix.py [hunk 4]
```diff
@@ -177,3 +182,1246 @@ def test_load_mcooler_format_fail():
...
+    # hic = hm.hiCMatrix()
+    # hic.nan_bins = []
+
+    # matrix = np.matrix([[np.nan for x in range(5)] for y in range(5)])
+
+    # hic.matrix = csr_matrix(matrix)
+    # hic.setMatrix(hic.matrix, cut_intervals)
+
+    # distance = hic.getCountsByDistance(mean=mean)
+
```
Comment (joachimwolff) [inline hicexplorer/test/test_hicmatrix.py]: Remove the out commented code.

File: hicexplorer/test/test_hicmatrix.py [hunk 5]
```diff
@@ -177,3 +182,1246 @@ def test_load_mcooler_format_fail():
...
+    hic.save_bing_ren(outfile)
+
+    # Test fails here due to __init__ of hiCMatrix
+    hm.hiCMatrix(outfile)
+
+    # test = hicTest.matrix
+
+    # nt.assert_equal(matrix.shape, test.shape)
+
```
Comment (joachimwolff) [inline hicexplorer/test/test_hicmatrix.py]: Remove out commented code.

File: hicexplorer/test/test_hicmatrix.py [hunk 6]
```diff
@@ -10,6 +10,8 @@
 from six import iteritems
 import pytest
 import logging
+from collections import OrderedDict
+
```
Comment (fidelram) [inline hicexplorer/test/test_hicmatrix.py]: This extra line is not necesary

**Comment by bgruening** [comment]
Cool beans @Rungetf!
I guess we should get this PR stable and not add new stuff to it, its already quite big :)

**Comment by joachimwolff** [comment]
Please merge latest commits to your fork and split the test cases in different stages that we overcome travis build time restriction. Moreover, please exchange the test data with the smaller test files I send you. Thanks a lot!

**Comment by bgruening** [comment]
@Rungetf any change we can get this in soon. It would be nice to have more testing in place before we merge bigger changes from @joachimwolff or potential hackathon outcomes next week.

**Comment by Rungetf** [comment]
@joachimwolff I split the tests into two folders `general/` and `long_run/` to add a `test long run` stage in the travis.yml. This file for sure can be optimized in terms of code quality, but I hope this will overcome the time out of 50 minutes that is set for travis. Long run Tests currently run in parallel during the `test long run` stage and should not take longer than 50 minutes each.

@joachimwolff Also testing of trivial runs of buildMatrix should now be much faster with the new `.bam` files I got from you.

General the merge conflict above should be occuring due to change of the `ROOT` directory in the test files because the `test_data` folder now is at a new relative position.

I'll try to find another solution for the problem with travis timing out after 50 minutes in the next days but I still hope this one will work for now after resolving the conflicts above.

**Comment by joachimwolff** [comment]
@Rungetf You have to resolve the conflicts to get travis running again.

**Comment by bgruening** [comment]
Does this example help? https://github.com/galaxyproject/galaxy/blob/dev/.travis.yml

**Comment by bgruening** [comment]
Muha!!! Thanks a bunch!

**Pull request merged**